src/utils/nodeTypeProcessing.ts

```diff
@@ -281,29 +287,39 @@ export function isStorageSpecificType(
   ) {
     visitedStructs.push(type.definition.id);
     return type.definition.vMembers.some((m) =>
-      isStorageSpecificType(safeGetNodeType(m, ast.compilerVersion), ast, visitedStructs),
+      isStorageSpecificType(safeGetNodeType(m, ast.inference), ast, visitedStructs),
     );
   }
   return false;
 }
 
-export function safeGetNodeType(node: Expression | VariableDeclaration, version: string): TypeNode {
+export function safeGetNodeType(
+  node: Expression | VariableDeclaration,
+  inference: InferType,
+): TypeNode {
   getContainingSourceUnit(node);
-  return getNodeType(node, version);
+  // if (node instanceof VariableDeclaration) {
+  //   return inference.variableDeclarationToTypeNode(node);
+  // }
+  // if (node instanceof TypeName) {
+  //   return inference.typeNameToTypeNode(node)
+  // }
+  // return inference.typeOf(node);
+  return getNodeType(node, inference);
```

Comment:
Eventually the commented code should be uncommented and the typestring parser removed

Reply:
Using InferType class to get node type yields 100+ failed tests right now so dealing with the 11 errors first seems to be a better strategy.